src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/Application.java

```diff
@@ -43,11 +46,9 @@ public static void main(String[] args) {
     public void run(String... args) {
         if (!runAppInErrorMode) {
             logger.info(String.format("%s started in normal processing mode", APPLICATION_NAME));
-            var mainCompletableFuture = loopingMainMessageConsumer.loopReadAndProcess();
-            var retryCompletableFuture = loopingRetryMessageConsumer.loopReadAndProcess();
-
-            // Wait until they are all done
-            CompletableFuture.allOf(mainCompletableFuture, retryCompletableFuture).join();
+            var retryThrottleMillis = retryThrottleSeconds * 1000L;
+            taskScheduler.scheduleWithFixedDelay(incomingMessageConsumer, 1L);
```

Comment:
just wondering i we might want > 1ms delay between consumes from main, would it potentially hog resources and prevent other threads from running?

Reply:
only one of these will be running at any given time thanks to fixed delay meaning it waits for the last execution to finish, then waits for the delay period, and then starts execution of the next one, and the task scheduler has two threads, and the main and retry consumers are the only things using those two threads. So it _shouldn't_ block anything.
But I can add a config value for this? The only requirement is it is > 0 as spring crashes in that case

Reply:
we could add config but it may confuse matters if we have one value in seconds and one in ms. Perhaps just increase it to 50 or 100ms (1/10th second)?  see what @sroberts2 thinks

Reply:
actually, too high a value will affect performance, maybe just leave it at 1 ad see how it performs